### PR TITLE
Bug 2079214: Switch default modal scroll behavior from inner modal body scroll to entire modal scroll to prevent bug and have consistent display of dropdown menus.

### PR DESCRIPTION
**Pull request by sg00dwin**
**TLDR;** Change the default modal scrolling behavior from inner content scroll  to scrolling of the entire modal view. Have consistent dropdown menu display behavior. And allow the modal at mobile sizes shrink based on its content size if less than viewport height, which matches PF behavior.
Fixes bug https://bugzilla.redhat.com/show_bug.cgi?id=2079214

Our custom modal component defaults to inner scrolling of the `modal-body` node activated by a `max-height: calc(100vh - 60px)`.  There is also an option, via `modal-content--no-inner-scroll`, added to allow `dropdown` menus to overlay the `modal-footer` and outside the modal if needed. But the combination of the two causes a bug when the modal content height exceeds the viewport height. https://bugzilla.redhat.com/show_bug.cgi?id=2079214

The PatternFly `<Modal>` overflow scroll behavior defaults to inner scrolling. But the PF `<Modal>` has conflicting aspects in that, dropdown menus are allowed to overlay outside the modal body. This can cause a bug where the menu items are inaccessible because the entire modal isn't scrollable in that scenario.  Upstream issue opened. https://github.com/patternfly/patternfly/issues/4957

So the question is should dropdown menus be contained within an inner modal scroll or can they overlay outside the content area and enable scrolling on the entire modal view itself? Based on the following, I made the choice to switch the default scrolling behavior from inner to entire modal and always overlay menus outside its body content container. 

1. The earlier decision to overlay dropdown menus outside the `modal-body` via `--no-inner-scroll` https://github.com/openshift/console/pull/1416
2. PF `<Modal>` displays dropdown menus outside of its modal content container.
3. Ideally if dropdown menus were contained with the `modal-body` inner scroll the dropdown should automatically switch the menu direction to up, when it is at the content bottom. Currently this functionally doesn't exist in the component.

Side note Bootstrap has both scroll options available. Entire modal  view scroll is the default behavior or inner scroll can be set optionally, in which case a dropdown menu at the content bottom switches direction to up. 

cc @jcaianirh @spadgett  

**Before bug fix**
cluster-bad

**After bug fix**
cluster-good

**Dropdown inner scroll**
tolerations-inner

---

**Dropdown entire scroll**
tolerations-outer

### CODE REVIEW

**Comment by rhamilto** [comment]
/approve

/lgtm

Wonderful explanation and nice work, @sg00dwin!
